changefeedccl: NPE in schema feed

```
panic: assignment to entry in nil map

goroutine 11112338 [running]:
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/schemafeed.(*schemaFeed).pauseOrResumePolling.func1.1(0x15a8f830?)
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/schemafeed/pkg/ccl/changefeedccl/schemafeed/schema_feed.go:467 +0x18d
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeedbase.(*Targets).EachTableID(0xc015a8f9d8?, 0xc015a8f8e8)
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeedbase/target.go:109 +0x8c
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/schemafeed.(*schemaFeed).pauseOrResumePolling.func1({0x18?, 0x40738738?, 0xc0?}, 0x16?)
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/schemafeed/pkg/ccl/changefeedccl/schemafeed/schema_feed.go:457 +0x88
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/schemafeed.(*schemaFeed).pauseOrResumePolling(0xc046f4c400, {0x72818a8?, 0xc02021ce00?}, {0x16?, 0x16?, 0x0?})
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/schemafeed/pkg/ccl/changefeedccl/schemafeed/schema_feed.go:487 +0x1a9
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/schemafeed.(*schemaFeed).peekOrPop(0xc046f4c400, {0x72818a8, 0xc02021ce00}, {0xc015a8fb88?, 0xd9eafa?, 0x0?}, 0x0)
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/schemafeed/pkg/ccl/changefeedccl/schemafeed/schema_feed.go:387 +0x74
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/schemafeed.(*schemaFeed).Peek(0xc015a8fb40?, {0x72818a8?, 0xc02021ce00?}, {0xc015a8fb58?, 0xd9f1cf?, 0x0?})
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/schemafeed/pkg/ccl/changefeedccl/schemafeed/schema_feed.go:372 +0x26
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvfeed.(*kvFeed).scanIfShould(0xc031f71040, {0x72818a8, 0xc02021ce00}, 0x1, 0x0, {0x8b6?, 0x13101e0?, 0xc0?})
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvfeed/kv_feed.go:376 +0x13a
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvfeed.(*kvFeed).run(0xc031f71040, {0x72818a8, 0xc02021ce00})
        github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvfeed/kv_feed.go:261 +0x145
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
        github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
        golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
        golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0xa5
```


Jira issue: CRDB-28773

--- a/pkg/ccl/changefeedccl/schemafeed/schema_feed.go
+++ b/pkg/ccl/changefeedccl/schemafeed/schema_feed.go
@@ -246,8 +246,6 @@ func (tf *schemaFeed) init() error {
 		return errors.AssertionFailedf("SchemaFeed started more than once")
 	}
 	tf.mu.started = true
-	tf.mu.allTableVersions1 = make(map[descpb.ID]descpb.DescriptorVersion)
-	tf.mu.allTableVersions2 = make(map[descpb.ID]descpb.DescriptorVersion)
 	return nil
 }
 
@@ -481,6 +479,11 @@ func (tf *schemaFeed) pauseOrResumePolling(
 	if atOrBefore.LessEq(tf.mu.highWater) {
 		// `atOrBefore` warrants a fast path already, with polling paused or not.
 		return atOrBefore, nil
+	}
+
+	if tf.mu.allTableVersions1 == nil {
+		tf.mu.allTableVersions1 = make(map[descpb.ID]descpb.DescriptorVersion)
+		tf.mu.allTableVersions2 = make(map[descpb.ID]descpb.DescriptorVersion)
 	}
 
 	// Always start with a stance to resume polling until we've proved otherwise.

